Refactored SkillObjectFactory to skill.model

KartikSuryavanshi opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR  fixes part of #10700 .
2. This PR does the following: Migrates SkillObjectFactory to a TypeScript class Skill and updates related code and tests to follow the new convention.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [ ] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [ ] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [ ] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [ ] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).



## Proof that changes are correct



## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

KartikSuryavanshi committed:
Refactored SkillObjectFactory to skill.model

KartikSuryavanshi committed:
made changes

KartikSuryavanshi committed:
made changes

KartikSuryavanshi committed:
made changes

HardikGoyal2003 commented on core/templates/components/question-directives/question-player/question-player-concept-card-modal.component.spec.ts:
```diff
@@ -67,7 +67,7 @@ describe('Question Player Concept Card Modal component', () => {
       declarations: [QuestionPlayerConceptCardModalComponent],
       providers: [
-        SkillObjectFactory,
+        Skill,
```
Why do we need this now in the providers array?

HardikGoyal2003 commented on core/templates/components/question-directives/questions-list/questions-list.component.spec.ts:
```diff
@@ -127,7 +126,7 @@ describe('Questions List Component', () => {
 
     ngbModal = TestBed.inject(NgbModal);
-    skillObjectFactory = TestBed.inject(SkillObjectFactory);
+    skill = TestBed.inject(Skill);
```
Why we need this?

HardikGoyal2003 commented on core/templates/components/question-directives/questions-list/questions-list.component.spec.ts:
```diff
@@ -89,10 +89,9 @@ describe('Questions List Component', () => {
   let contextService: ContextService;
   let questionValidationService: QuestionValidationService;
-  let skillObjectFactory: SkillObjectFactory;
+  let skill: Skill;
```
Why we need this?

HardikGoyal2003 commented on core/templates/components/state-editor/state-skill-editor/state-skill-editor.component.spec.ts:
```diff
@@ -65,7 +65,7 @@ describe('State Skill Editor Component', () => {
   let userService: UserService;
   let skillBackendApiService: SkillBackendApiService;
-  let skillObjectFactory: SkillObjectFactory;
+  let skill: Skill;
```
Ditto as above and for all the test files below

KartikSuryavanshi replied:
Done

HardikGoyal2003 commented on core/templates/domain/skill/skill-backend-api.service.ts:
```diff
@@ -72,7 +68,7 @@ interface SkillAssignmentForDiagnosticTestBackendResponse {
   constructor(
     private http: HttpClient,
-    private skillObjectFactory: SkillObjectFactory,
+    private skill: Skill,
```
Why do we need this? As it is no more a service

HardikGoyal2003 commented on core/tests/root-files-config.json:
```diff
@@ -17,7 +17,7 @@
     "extensions/rich_text_components/richTextComponentsRequires.ts",
     "core/templates/domain/exploration/LostChangeObjectFactorySpec.ts",
-    "core/templates/domain/skill/SkillObjectFactorySpec.ts",
+    "core/templates/domain/skill/skill.model.spec.ts",
```
Remove this line

HardikGoyal2003 commented on core/templates/services/angular-services.index.ts:
```diff
@@ -795,7 +795,7 @@ export const angularServices: [string, Type<{}>][] = [
   ['SkillEditorRoutingService', SkillEditorRoutingService],
   ['SkillMasteryBackendApiService', SkillMasteryBackendApiService],
-  ['SkillObjectFactory', SkillObjectFactory],
+  ['Skill', Skill],
```
Remove this line

HardikGoyal2003 commented on core/templates/services/angular-services.index.ts:
```diff
@@ -100,7 +100,7 @@ import { PretestQuestionBackendApiService } from 'domain/question/pretest-questi
 import { ExplorationRecommendationsBackendApiService } from 'domain/recommendations/exploration-recommendations-backend-api.service';
 import { PracticeSessionsBackendApiService } from 'pages/practice-session-page/practice-session-backend-api.service';
-import { SkillObjectFactory } from 'domain/skill/SkillObjectFactory';
+import { Skill } from 'domain/skill/skill.model';
```
Remove

HardikGoyal2003 commented on core/templates/pages/topics-and-skills-dashboard-page/modals/create-new-skill-modal.component.ts:
```diff
@@ -61,14 +62,24 @@ export class CreateNewSkillModalComponent {
     private skillEditorStateService: SkillEditorStateService,
-    private skillObjectFactory: SkillObjectFactory,
-    private changeDetectorRef: ChangeDetectorRef
+    private skill: Skill,
```
Why do we need it?

KartikSuryavanshi committed:
made changes

KartikSuryavanshi committed:
Made changes

KartikSuryavanshi committed:
Merge branch 'develop' into refactor-skill-object-factory-to-model

KartikSuryavanshi committed:
Merge branch 'develop' into refactor-skill-object-factory-to-model

KartikSuryavanshi committed:
Merge branch 'develop' into refactor-skill-object-factory-to-model

KartikSuryavanshi committed:
Merge branch 'develop' into refactor-skill-object-factory-to-model

mon4our committed:
[GSoC 2025] M1.7 - Fix part of #18305: Created the frontend and backend for the Subtopics to support Study Guides (#22798)

* Made fe changes for subtopic editor

* Made changes to Create new subtopic modal and added section mat card

* Made changes for the subtopic editor to support study guides

* Lint fixes

* Lint fixes

* Type fixes

* Backend nits, moving files

* Added acceptance test for study guides

* Acceptance test modifications and screenshots

* Nit

* study guide, sections frontend model changes

* Addressed some comments

* Addressed all comments

* Updated common.py with new acceptance test name

* Linting fix and nit

* Fixed frontend tests for create new subtopic modal component

* Added frontend tests for add study guide section component

* Added another test for add study guide section component

* Nit

* Added backend tests for EditableStudyGuideDataHandler

* Acceptance test fixes

* Addressed comments

* Ran prettier

* Fixed frontend tests

* Added frontend tests

* Frontend coverage

* Topic editor coverage

* Study guide domain coverage

* Renamed variables

* Study guide domain coverage

* Complete frontend coverage

* Made topic services match M1.6

* Addressed comments

* Fixed backend tests

* Fixed backend nits

* Fixed tests

* Fixed backend tests

* Added acceptance test screenshots

* Fixed type error

* Addressed comments

* Addressed comments

* Fixed tests

* Lint fixes

* Lint fixes

* Lint fixes

* Fixed acceptance test

* Updated screenshot

* Updated screenshot

Helper2020 committed:
Exp services test time fix (#22835)

* yarn fix suggestion from oppia discussion board

* Fix test_get_last_updaeted_by_human

* fix typo

* Addresed comment

jayam04 committed:
Removes flaky steps in Acceptance tests. (#22837)

* Removed flaky test steps

* Removed publication part.

* Lint Fixes.

* Removed steps with flaky manage goals.

* Lint Fixes

U8NWXD committed:
Fix #19110: Implement job to copy mis-placed translation images (#21378)

* Implement job to copy mis-placed translation images

* Implement job to copy mis-placed translation images

* Fix type annotations in JobRunResult

The as_stdout() and as_stderr() functions were annotated as only
accepting str and int arguments, but they are designed (and frequently
used) to process arbitrary input types. Therefore, they should actually
accept inputs of type Any.

* Add tests

* Fix tests

* Fix job imports test

* Separate debugging and audit jobs

* Clean up validation logic

* Add logging

* Document exception

* Handle translation_html values of type List[str]

* Update debugging check to allow for empty translation_html

* Fix lint

* Add debugging for src_exist unpacking error

* Handle repeated references to the same image files among translation suggestions

* Remove debugging code

* Avoid dependency on NDB in get_application_id

* Fix type and lint errors

---------

Co-authored-by: Sean Lip <sean@seanlip.org>
Co-authored-by: Nikhil <nikhil.agarwal.2019@gmail.com>

mon4our committed:
Fixed Bug in M1.7 #22798: Newly added Acceptance Tests failing due to screenshot mismatch (#22856)

* Removed prod mobile screenshots

* Added code to download screenshots

* Added code to download screenshots

* Added code to download screenshots

* Added code to download screenshots

* Re established status quo

U8NWXD committed:
Revert wiki changes from web interface by force-pushing from oppia-web-developer-docs (#22783)

Make wiki revert workflow force-push

tiagombrbrito committed:
Fix #20893: Prevent fast-click events for newsletter subscription (#22664)

* Fix fast-click events for newsletter subscription

- Fix splash page ngx-toastr CSS import
- Add boolean subscriptionProcessing to OppiaFooterComponent
- Add function disableNewsletterSubscription to OppiaFooterComponent
- Update function subscribeToMailingList in OppiaFooterComponent
- Add front-end tests for disableNewsletterSubscription
- Update front-end tests for subscribeToMailingList

* Address review comments

- Add i18n variable for newsletter warning
- Add ngModelChange for newsletter input
- Add CSS for oppia-footer-newsletter-warning
- Remove delay from subscribeToMailingList
- Add boolean emailDuplicated to OppiaFooterComponent
- Add set emailsSubscribed to OppiaFooterComponent
- Add function isAlreadySubscribed to OppiaFooterComponent
- Add function clearNewsletterWarning to OppiaFooterComponent
- Update front-end tests for subscribeToMailingList
- Add front-end tests for isAlreadySubscribed
- Add front-end test for clearNewsletterWarning

* Fix add email to set

* Fix emailAddress type handling

---------

Co-authored-by: Mohit Ruwatia <120438312+mon4our@users.noreply.github.com>

HardikGoyal2003 committed:
[GSoC 2025] M 1.3.2 - Fix part of #19217: Deprecated the explorationPlayerStateService (#22810)

* Refactored the folder structure

* Fix imports

* Fix some more imports

* Fix some more imports

* Fix some more imports

* Fix folder structure

* Fix some more imports

* Fix some type errors

* Fix some more imports

* Fix type errors for Question Engine file

* Fix ts checks excluded file paths

* Fix file imports

* Dismantled explorationPlayerStateService

* Fixed CheckpointProgressService

* Fixed ProgressUrlService

* Fixed some more files

* Removed ExplorationPlayerStateService

* lint fix

* Fixed fileoverview

* Renamed Context-Service

* lint fix

* Fix frontend tests

* Removed version from explorationModeService

* Added CurrentEngineService

* Added Frontend tests for the new services

* Fix some frontend tests

* Fixed some services

* Added ExplorationInitializationService

* lint fix

* Fix import paths

* Fix some type errors

* Added ExplorationInitializationService in exclusion list

* Fix imports error

* Fix frontend tests

* Fix Pid null bug

* Fix frontend tests

* Fix frontend tests

* Fix E2E tests

* Fix version state

* Fix frontend tests and coverage

* Fix type errors

* Fix some more tests

* Fix some more tests

* Fixed frontend test coverage

* Fix some more tests

* Fix null instances

himkar-singh committed:
Refactored jQuery constructs and replace with native JavaScript and CSS for animations (#22847)

* Commented out problematic code for e2e test

* Attempting to fix failing e2e test on ci

* Attempting to fix failing ci checks

* Attempting to fix failing ci checks

* Fixed failing coverage checks

* Attempting to fix failing ci check

* Attempting to fix failing ci check

* Fixed failing ci check

* Fixed failing ci checks

* Removed unnecessary test

* Fixed failing linter checks

* Fixed mobile save button issue

* Trying to fix mobile navbar dropdown issue

* Fixed mobile dropdown issue

* Fixed failing test

* Fixed failing lint checks

* Fixed failing lint checks

* Ran prettier

* Fixed error in console

* Ignored error in console due to third party lib

* Fixed failing test

* Allowed a console error to fix test

* Ignored all console errors in exploration editor

* Tried to fix failing test

* Made requested changes

* Fixed failing e2e test

* Fixed Failing Test

* Fixed failing test

* Fixed LostChangesModal Issue

* Tried to fix failing e2e test

* Fixed failing e2e test

* Fixed failing e2e test

* Tried to fix failing test

* Tried to fix failing test

* Removed unnecessary loading of script

* Adding the needed scripts

* Fixed mobile button issue

* Fixed linting issues

* Fixed failing tests

* Fixed rte issue

* Fixed a failing test

* Fixed linting issue

* Trying to fix failing test

* Tried to fix failing test

* Trying to fix failing acceptance test

* Fixed failing acceptance test

* Added scripts for guppy

* Enabled video recording

* Enabled video recording

* Added logging for ci

* Enabled all videos to be recorded

* Removed semicolon from e2e test workflow file

* Fixed CI flake

* test commit

* upgraded python to 3.9

* Fixed failing acceptance test

* Added logging

* Added Logging

* Getting the CI variable name

* Flaky test gets skipped in the CI

* Enabled all video recording

* Fixed history tab button

* Fixed failing test

* Edited comment

* Fixed failing test

* Enabled video recording

* Tried to fix failing e2e test

* Made requested changes

* Ran prettier

* Trying to fix failing e2e test for state name input

* Trying to fix failing e2e test for state name input

* Enabled video recording

* Trying to fix state name input e2e flake

* Tried to fix failing e2e test for state name input

* Fixed linting issues

* Trying to fix state name input e2e flake

* Trying to fix state name input e2e flake

* Trying to fix state name input e2e flake

* Trying to fix state name input e2e flake

* Tried to fix failing tests

* Updated variable names and added comments in insert-script-service

* Changed function name to reflect mobile mode

* Removed unnecessary code

* Ran prettier

* Updated variable name in insert script service spec file

* Ran prettier

* Trying to fix failing e2e test

* Removed unnecessary lines from test

* Made requested changes

* Enabled all videos to be recorded

* Fixed failing e2e test

* Fixed failing e2e test

* Disabled video recording

* Ran prettier

* Removed unnecessary code

* Fixed flaky e2e test

* Added comment

* Added console logs to debug acceptance test

* Attempting to fix failing e2e test

* Fixed e2e test

* Ran prettier

* Removed unnecessary code

* Ran prettier

* Fixed failing frontend test

* Created paths for angular router

* Fixed failing e2e test

* Removed console log

* Removed logs and testing code

* Removed logs and testing code

* Removed unnecessary code

* Code refactoring

* Fixed failing e2e test

* Fixed failing acceptance test

* Ran prettier

* Made requested changes

* Removed console logs

* Added comment

* Tried to fix failing e2e test

* Removed sharedcompmodule from app.module

* Fixed linting error

* Fixed failing e2e test

* Fixed e2e test

* Fixed e2e test

* Fixed e2e test

* Tried to fix acceptance test

* Tried to fix failing acceptance test

* Tried to fix acceptance test

* Tried to fix acceptance test

* Tried to fix acceptance test

* Tried to fix acceptance test

* Tried to fix acceptance test

* Tried to fix acceptance test

* Linting

* Acceptance test

* Acceptance Test

* Acceptance Test

* Acceptance test

* Acceptance test

* Acceptance Test

* Acceptance Test

* Fixed Acceptance Test

* Removed unnecessary code

* Taking screenshots

* Taking screenshots

* Taking screenshot

* Taking screenshot

* Taking screenshots

* Taking screenshot

* Removed testing code

* Take ss

* Trying to take screenshot

* Taking screenshot to debug

* Fixed path for ss upload

* Removed screenshot method

* Taking screenshot

* Tried to fix acceptance test

* Taking screenshot

* Ran prettier

* Removed screenshot

* Taking screenshot

* Tried to fix acceptance test

* Logging active element

* Logging active element

* Logging active element

* Log active element

* Log active element

* Uncommented code

* Running original code for testing

* Tried to fix acceptance test

* Fixed acceptance test

* Ran prettier

* Testing hypothesis

* Testing hypothesis

* Uncommented code

* Fixed test

* logging focus

* logging focus

* Added timeout

* Added wait for network to become idle

* Removed leftover code from before merging mathjax pr

* Nit

* Addressed comments

* Addressed comments

* Nits and pencilcode script

* Removed unnecessary code, properly loaded pencilcode script

* fix node error

* fix error and added some code

* remove frontend error

* added test to fix coverage issue

* fix error

* fix errro 2

* fix error

---------

Co-authored-by: Mohit Ruwatia <mohitruwatia2312@gmail.com>
Co-authored-by: Mohit Ruwatia <120438312+mon4our@users.noreply.github.com>
Co-authored-by: Hardik Goyal <135348086+HardikGoyal2003@users.noreply.github.com>
Co-authored-by: Himkar_Singh <himkar-singh@M.X-vT>

chris7716 committed:
Adds new translation opportunity model structure (#22816)

* service file name changed to backend-api

* backend api file added

* previous service file removed from CODEOWNERS

* new backend service file added to  CODEOWNERS

* Python file change

* Rollback

* Add new feature flag

* Add frontend change

* First pass

* Fix lint issues

* Fix lint

* Fix lint

* Fix lint

jayam04 committed:
[GSoC 2025 - Acceptance Tests] M1.1 - Add pre/post checks for logged-in-user.ts (#22759)

* First check

* Revert It: Removed some of the workflow tests and updated full stack tests

* Revert "First check"

This reverts commit bdac59142b7d70fc488b4ee5a29acb51b98bf8bd.

* Changes upto line 498 in logged-in-user.ts

* Revert It: removes log level in test to be less verbose

* Added more pre/post checks until line 693

* Added more pre/post checks until line 1024

* Patch A: Fix Incorrect Selector

* Patch B: Primarily added pre-post checks for redirect of URL

* Fixed incorrect import

* Fixed incorrect import -- 2

* Added more pre/post checks until the last line

* Patch C

* Patch D

* More fixes

* Revert "Revert It: removes log level in test to be less verbose"

This reverts commit 623ce234c9b8599e9fa85484805396186d9b6936.

* Revert "Revert It: Removed some of the workflow tests and updated full stack tests"

This reverts commit 7ac694dcd3a38b208b3702171421cbdc47aa3486.

* Fixed test file, so test can run

* Fixed Lints.

* Changed acceptance-test- to e2e-test- for cleaner naming

* Refactored function names

* Added descriptions

* Fixed failing post-check

* Removed TODO comments

jayam04 committed:
[GSoC 2025 - Acceptance Tests] M1.5 - Add pre/post checks for all of the users in puppeteer acceptance tests. (#22809)

* First pre-check

* Added pre/post checks upto line 527

* REVERT IT: Removes all tests from CI except acceptance tests.

* Added pre/post checks upto line 1140

* Fixed Issues with commit: (Add pre/post checks upto line 527)

* Added pre/post checks upto line 2333

* Fixed Issues with commit: (Add pre/post checks upto line 1140)

* Added pre/post checks upto the last line

* Fixed all issues.

* Fixed issues with mobile viewport

* Added pre/post checks for blog-admin, blog-post-editor

* Added pre/post checks for moderator

* Added pre/post checks for Question Submitter and Question Admin

* Added pre/post checks for superAdmin

* Added pre/post checks for all of the remaining users

* Fixes.

* Removed unused commented code

* Fixed failing tests. -- 1

* Fix to failing blog post tests.

* Revert "REVERT IT: Removes all tests from CI except acceptance tests."

This reverts commit e59549ed61cb08c0159b8bb27c3539aa1a3b9716.

* Lint Fixes

* Lint Fixes

* Fixes

* Lint Fixes

* removed all trailing spaces in story-editor.component.html

* Fixed failing misc-...

* Addressed Comments.

HardikGoyal2003 committed:
[GSoC 2025] M 1.4 - Fix part of #19217: Migrate states from the conversation-skin component (#22853)

* Refactored the folder structure

* Fix imports

* Fix some more imports

* Fix some more imports

* Fix some more imports

* Fix folder structure

* Fix some more imports

* Fix some type errors

* Fix some more imports

* Fix type errors for Question Engine file

* Fix ts checks excluded file paths

* Fix file imports

* Dismantled explorationPlayerStateService

* Fixed CheckpointProgressService

* Fixed ProgressUrlService

* Fixed some more files

* Removed ExplorationPlayerStateService

* lint fix

* Fixed fileoverview

* Renamed Context-Service

* lint fix

* Fix frontend tests

* Removed version from explorationModeService

* Added CurrentEngineService

* Added Frontend tests for the new services

* Fix some frontend tests

* Fixed some services

* Added ExplorationInitializationService

* lint fix

* Fix import paths

* Fix some type errors

* Added ExplorationInitializationService in exclusion list

* Fix imports error

* Fix frontend tests

* Fix Pid null bug

* Fix frontend tests

* Fix frontend tests

* Fix E2E tests

* Fix version state

* Fix frontend tests and coverage

* Fix type errors

* Fix some more tests

* Fix some more tests

* Fixed frontend test coverage

* Fix some more tests

* Removed duplicate logic

* Shifted some states

* Removed allowlist collection logic and refactored stuck logic

* Removed obselete code

* Migrated nextCard state and added some JsDocs

* Migrated mostRecentlyReachedCheckpoint state and refactored some functions

* Fix type checks

* Fix frontend tests

* Fix null instances

* Fix frontend tests

* Fix frontend tests converage

* Fix E2E test

mon4our committed:
[GSoC 2025] M1.12 - Fix part of #19851: Added worked examples feature flag (#22860)

Added worked examples feature flag

mon4our committed:
[GSoC 2025] M1.8 - Fix part of #18305: Created the frontend and backend for the study guide viewer and preview tabs (#22848)

* Made fe changes for subtopic editor

* Made changes to Create new subtopic modal and added section mat card

* Made changes for the subtopic editor to support study guides

* Lint fixes

* Lint fixes

* Type fixes

* Backend nits, moving files

* Added acceptance test for study guides

* Acceptance test modifications and screenshots

* Nit

* study guide, sections frontend model changes

* Addressed some comments

* Addressed all comments

* Updated common.py with new acceptance test name

* Linting fix and nit

* Fixed frontend tests for create new subtopic modal component

* Added frontend tests for add study guide section component

* Added another test for add study guide section component

* Nit

* Added backend tests for EditableStudyGuideDataHandler

* Acceptance test fixes

* Addressed comments

* Ran prettier

* Fixed frontend tests

* Added frontend tests

* Frontend coverage

* Topic editor coverage

* Study guide domain coverage

* Renamed variables

* Study guide domain coverage

* Complete frontend coverage

* Made topic services match M1.6

* Addressed comments

* Fixed backend tests

* Fixed backend nits

* Fixed tests

* Fixed backend tests

* Added acceptance test screenshots

* Fixed type error

* Created subtopic preview

* Created study guide viewer

* nit

* Added buttons

* Centered button

* Added navigation to button

* Updated screenshot

* Linting fix

* Type fixes

* Type and lint fixes

* Type and lint fixes

* Nits

* Restored original screenshots

* Fixed I18N keys

* Fixed backend test

* Fixes

* Added tests

* Frontend coverage

* Added backend tests

* Added acceptance test

* Fixes

* Test fixes

* Taking screenshots from CI

* Frontend coverage

* Fixed acceptance test

* Frontent coverage

* Fixed acceptance test

* Fixed acceptance test

* Fixed acceptance test

* Fixed acceptance test

* Added screenshot

* Fixed acceptance test

* Taking screenshots

* Taking screenshots

* Nit

* Backend coverage

* Added test to common.py

* Nit

* Removed desktop screenshot functionality

* Finalized PR

* Fixed bug

* Renamed helper method for acceptance tests

mon4our committed:
[GSoC 2025] M1.9 - Fix part of #18305: Updated the Android controller to use the new structure. (#22870)

* Updated android controller

* Lint fixes

* Addressed comments

* Fixed lint error

jayam04 committed:
[GSoC 2025 - Acceptance Tests] M1.2.0 - Updates common.py to use acceptance.json for test suites. (#22871)

* Updates common.py to use acceptance.json for test suites.

* Lint fixes

* Addressed few comments.

* Addressed review comments -- 0

HardikGoyal2003 committed:
[GSoC 2025] M 1.5 - Fix part of #19217: Migrate navigation logic from the conversation-skin component (#22862)

* Refactored the folder structure

* Fix imports

* Fix some more imports

* Fix some more imports

* Fix some more imports

* Fix folder structure

* Fix some more imports

* Fix some type errors

* Fix some more imports

* Fix type errors for Question Engine file

* Fix ts checks excluded file paths

* Fix file imports

* Dismantled explorationPlayerStateService

* Fixed CheckpointProgressService

* Fixed ProgressUrlService

* Fixed some more files

* Removed ExplorationPlayerStateService

* lint fix

* Fixed fileoverview

* Renamed Context-Service

* lint fix

* Fix frontend tests

* Removed version from explorationModeService

* Added CurrentEngineService

* Added Frontend tests for the new services

* Fix some frontend tests

* Fixed some services

* Added ExplorationInitializationService

* lint fix

* Fix import paths

* Fix some type errors

* Added ExplorationInitializationService in exclusion list

* Fix imports error

* Fix frontend tests

* Fix Pid null bug

* Fix frontend tests

* Fix frontend tests

* Fix E2E tests

* Fix version state

* Fix frontend tests and coverage

* Fix type errors

* Fix some more tests

* Fix some more tests

* Fixed frontend test coverage

* Fix some more tests

* Removed duplicate logic

* Shifted some states

* Removed allowlist collection logic and refactored stuck logic

* Removed obselete code

* Migrated nextCard state and added some JsDocs

* Migrated mostRecentlyReachedCheckpoint state and refactored some functions

* Fix type checks

* Fix frontend tests

* Fix null instances

* Fix frontend tests

* Fix frontend tests converage

* Migrated back and forth logic

* Added docstring anf moved some logic

* Migrated visitedCheckpointStateNames

* Removed inStoryMode binding

* Migrated some more states

* Fix E2E test

* Migrate some more states

* Migrated some more states

* Migrated some more states

* Created ChapterProgressService

* Fix ts checks and styling

* Added tests for ChapterProgressService

* Fix frontend tests

* Lint fix

* Fix frontend tests

* Fix frontend tests

* Fix frontend tests

* Fix frontend tests

* lint fix

* Fix acceptance tests

* Addressed PR comments

* Addressed PR comments

sagueso committed:
Fix part of #10700: Refactor of SolutionObjectFactory (#22657)

* refactor solution of

* fix import error

* fix specs providers

* delete refrences from root-files-config.json

* removed unused import

* Refactor short answer response handling: move logic to ResponsesService and update uses

* Refactor ExplorationHtmlFormatterService dependency

* unused provider

* fixed imports

* fix test

* fix failing test

HardikGoyal2003 committed:
[GSoC 2025] M 1.6 - Fix part of #19217: Finalize exploration engine service (#22881)

* Refactored the folder structure

* Fix imports

* Fix some more imports

* Fix some more imports

* Fix some more imports

* Fix folder structure

* Fix some more imports

* Fix some type errors

* Fix some more imports

* Fix type errors for Question Engine file

* Fix ts checks excluded file paths

* Fix file imports

* Dismantled explorationPlayerStateService

* Fixed CheckpointProgressService

* Fixed ProgressUrlService

* Fixed some more files

* Removed ExplorationPlayerStateService

* lint fix

* Fixed fileoverview

* Renamed Context-Service

* lint fix

* Fix frontend tests

* Removed version from explorationModeService

* Added CurrentEngineService

* Added Frontend tests for the new services

* Fix some frontend tests

* Fixed some services

* Added ExplorationInitializationService

* lint fix

* Fix import paths

* Fix some type errors

* Added ExplorationInitializationService in exclusion list

* Fix imports error

* Fix frontend tests

* Fix Pid null bug

* Fix frontend tests

* Fix frontend tests

* Fix E2E tests

* Fix version state

* Fix frontend tests and coverage

* Fix type errors

* Fix some more tests

* Fix some more tests

* Fixed frontend test coverage

* Fix some more tests

* Removed duplicate logic

* Shifted some states

* Removed allowlist collection logic and refactored stuck logic

* Removed obselete code

* Migrated nextCard state and added some JsDocs

* Migrated mostRecentlyReachedCheckpoint state and refactored some functions

* Fix type checks

* Fix frontend tests

* Fix null instances

* Fix frontend tests

* Fix frontend tests converage

* Migrated back and forth logic

* Added docstring anf moved some logic

* Migrated visitedCheckpointStateNames

* Removed inStoryMode binding

* Migrated some more states

* Fix E2E test

* Migrate some more states

* Migrated some more states

* Migrated some more states

* Created ChapterProgressService

* Fix ts checks and styling

* Added tests for ChapterProgressService

* Fix frontend tests

* Lint fix

* Fix frontend tests

* Fix frontend tests

* Fix frontend tests

* Fix frontend tests

* lint fix

* Removed unused functions

* lint fix

* Fix acceptance tests

* lint fix

* lint fix

* Added some DocString

* Removed unused function and added some more docString

* Fix frontend tests

* Fix frontend tests

* Fix frontend test coverage

* Fix frontend test coverage

* lint fix

* Fix some merging issues

* Addressed PR comments

KartikSuryavanshi committed:
fixed merge conflicts
